### ISSUE TITLE
Relative humidity values can exceed 100%

### ISSUE THREAD

**Issue by cloneofghosts** [Pirate-Weather/pirateweather#360]
### Describe the bug

As shown in [this comment](https://github.com/Pirate-Weather/pirate-weather-ha/issues/330#issuecomment-2439904606) relative humidity can exceed 100% when it should not be possible to do so.

### Expected behavior

Humidity should not be able to exceed 100%

### Actual behavior

Humidity can exceed 100%

### API Endpoint

Production

### Location

Romania

### Other details

I would also check to make sure relative humidity cannot go below 0% and other data points to make sure they have a minimum/maximum value where it makes sense. An example would be to make sure cloud cover cannot exceed 1 or go below 0.

### Troubleshooting steps

- [X] I have searched this repository and [Home Assistant Repository](https://github.com/alexander0042/pirate-weather-ha) to see if the issue has already been reported.
- [X] I have read through the [API documentation](https://pirateweather.net/en/latest/API/) before opening this issue.
- [X] I have not included my API key anywhere in my issue.
- [X] I have written an informative title.

**Comment by cloneofghosts** [comment]
@alexander0042 Not a huge rush on this one but curious if this is something that's being worked on currently? I see that the dev endpoint has been down since yesterday with this message:

```
{
    "message": "An invalid response was received from the upstream server"
}
```

**Comment by cloneofghosts** [comment]
@alexander0042 Noticed this morning with the extreme temperatures issue that `cloudCover` can go above 100%. A location to test with is London, UK (51.50853,-0.12574). You'll have to extend out the hourly forecast and the first occurrence I see is at 1731880800.

**Comment by alexander0042** [comment]
You'd think that I'd have come across np.clip before, but this is the first time I've noticed it! I have clipping setup for most of the hourly values, but need to go through and add it to the currently ones. I'll keep this tagged for 2.4 so it makes it in

**Comment by cloneofghosts** [comment]
@alexander0042 Any chance this will get fixed for V2.4? You mentioned that you want to get this fixed for V2.4 but this hasn't seemed to be fixed as far as I can tell.

**Comment by alexander0042** [comment]
Ok, just pushed this one! np.clip is a really clean function, so happy to have it in there!

**Comment by cloneofghosts** [comment]
@alexander0042 Not sure if this is caused by this change or something else but I'm noticing on the dev endpoint that the temperature and dew point in the hourly/daily blocks is fixed at -213.15. Everything else seems to be normal though.

**Comment by alexander0042** [comment]
I'm also getting it on temperature and apparent temperature, so fixing it now! The -213 comes from the new min level, which I capped at -90 C

**Comment by alexander0042** [comment]
That was easy- I forgot that it uses K for internal units, not C, so the clipping was off. Corrected it now!

**Comment by cloneofghosts** [comment]
@alexander0042 I noticed some other issues this morning and not sure if they're related:

1. Current pressure is always 11
2. Elevation is always 0
3. Something weird is going on with current `windSpeed` or current `windGust`. Maybe only in HRRR domain? This is what I see for my location:

```
  "windSpeed": 2.54,
  "windGust": 29.35,
```

I checked a few other locations in the HRRR domain and they're seeing similar things. Maybe the `windSpeed` is being shown in m/s and not km/h?
